src/main/java/interpreter/Interpreter.java

```diff
@@ -215,17 +223,19 @@ private static void validateDemo(Demo demo) {
         if (new HashSet<>(nodes).size() != nodes.size()) {
             throw new InterpreterException("Duplicate nodes");
         }
-        for (Edge edge: edges) {
-            if (!nodes.contains(edge.getStart())) {
-                throw new InterpreterException("Node " + edge.getStart() + " does not exist");
+        if (demo.getAlgorithm() != Algorithm.NOTHING) {
```

Comment:
the edges should still be validated for the nothing algorithm

Reply:
ok right, i will the forloop out of the if statement.

Reply:
i will move the forloop out of the if statement.